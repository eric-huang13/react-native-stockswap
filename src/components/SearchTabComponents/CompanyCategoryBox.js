import React, {Component} from 'react';
import {Text, View, StyleSheet, SafeAreaView} from 'react-native';
import {moderateScale} from '../../util/responsiveFont';

export default class CompanyCategoryBox extends Component {
  render() {
    const {item, index} = this.props;

    return (
      <SafeAreaView
        style={
          index % 2 !== 0
            ? {...style.container, backgroundColor: '#2a334a'}
            : {...style.container}
        }>
        <View key={item.id} style={style.companyRow}>
          <View style={style.nameContainer}>
            <Text style={style.symbol}>{item.ticker}</Text>
            <Text style={style.title}>{item.title}</Text>
          </View>
          <View style={style.detailsContainer}>
            <Text style={style.price}>${item.quote.volumeWeightedAveragePrice}</Text>
            <Text style={style.percentage}>{item.change}%</Text>
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

const style = StyleSheet.create({
  container: {
    backgroundColor: '#324165',
    paddingVertical: moderateScale(4),
  },
  companyRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: moderateScale(12),
    alignContent: 'space-between',
    marginVertical: moderateScale(6),
  },
  nameContainer: {
    flexDirection: 'column',
    justifyContent: 'space-evenly',
  },
  detailsContainer: {
    marginTop: moderateScale(5),
  },
  symbol: {
    fontFamily: 'Montserrat-Bold',
    color: '#FFFFFF',
    fontSize: moderateScale(20.5),
  },
  title: {
    color: '#d0d3dc',
    marginTop: moderateScale(3),
    fontFamily: 'Montserrat-Regular',
    fontSize: moderateScale(14),
  },
  price: {
    color: 'white',
    fontSize: moderateScale(22),
    fontWeight: 'bold',
    fontFamily: 'Montserrat-Bold',
    color: '#FFFFFF',
    fontSize: moderateScale(20.5),
  },
  percentage: {
    color: '#5fc48c',
    alignSelf: 'flex-end',
    marginTop: moderateScale(3),
    fontFamily: 'Montserrat-Medium',
    fontSize: moderateScale(14),
  },
});
