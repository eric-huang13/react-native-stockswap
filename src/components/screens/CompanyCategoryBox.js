import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'


export default class CompanyCategoryBox extends Component {
    render() {
        const {item} = this.props;
    
        return (
            <View style={style.container}>
            <View key={item.id} style={style.companyRow}>
                <View style={style.nameContainer}>
              <Text style={style.symbol}>{item.symbol}</Text>
              <Text style={style.title}>{item.title}</Text>
              </View>
              <View style={style.detailsContainer}>
              <Text style={style.price}>${item.price}</Text>
              <Text style={style.percentage}>{item.percentage}</Text>
              </View>
            </View>
            </View>
        );
      }
    }

    const style = StyleSheet.create({
        container:{
       
        },
        searchInputContainer: {
          alignItems: "center",
          padding: 4.8,
          marginTop: 0,
        },
        searchInput: {
          borderWidth: 1,
          width: "90%",
          fontSize: 15,
          fontStyle:"italic",
          textAlign: "center",
          padding: 2.5,
          opacity: 0.8,
          marginBottom:20
        },
        header: {
          fontSize: 18.5,
          fontWeight: "700",
          color: "rgb(123, 123, 124)",
          marginLeft: 23,
          marginTop: 8.5,
        },
        companyRow:{
           
          flexDirection:"row",
          justifyContent:"space-between",
          paddingHorizontal:12,
          alignContent:"space-between",
          marginVertical:8
         
        },
        nameContainer:{
          flexDirection:"column",
        },
        detailsContainer:{
       
        },
        symbol:{
            fontSize:20,
      fontWeight:"bold",
       
      },
        title:{
          color:"grey"
        },
          price:{
              fontSize:20,
       
        },
        percentage:{
          color:"grey"
       
        },
      });
      