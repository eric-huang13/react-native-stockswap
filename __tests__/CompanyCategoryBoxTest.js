import React from 'react';
import CompanyCategoryBox from '../src/components/SearchTabComponents/CompanyCategoryBox';
import ShallowRenderer from 'react-test-renderer/shallow'; 


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<CompanyCategoryBox item={{id:'test' }}/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});