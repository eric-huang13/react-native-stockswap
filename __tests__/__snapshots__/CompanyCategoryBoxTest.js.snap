// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "paddingVertical": 5.649635036496351,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "space-between",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 8.474452554744525,
        "paddingHorizontal": 16.94890510948905,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-evenly",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 28.954379562043794,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#d0d3dc",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 19.77372262773723,
            "marginTop": 4.237226277372263,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 7.062043795620438,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 28.954379562043794,
            "fontWeight": "bold",
          }
        }
      >
        $
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#5fc48c",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 19.77372262773723,
            "marginTop": 4.237226277372263,
          }
        }
      />
    </View>
  </View>
</ForwardRef(SafeAreaView)>
`;
